ocaml/xapi/message_forwarding.ml

```diff
@@ -1609,14 +1609,14 @@ module Forward = functor(Local: Custom_actions.CUSTOM_ACTIONS) -> struct
     let pool_migrate ~__context ~vm ~host ~options =
       info "VM.pool_migrate: VM = '%s'; host = '%s'"
         (vm_uuid ~__context vm) (host_uuid ~__context host);
-      if Helpers.rolling_upgrade_in_progress ~__context
-      then begin
-        let source_host = Db.VM.get_resident_on ~__context ~self:vm in
-        Helpers.assert_host_versions_not_decreasing
-          ~__context
-          ~host_from:(Helpers.LocalObject source_host)
-          ~host_to:(Helpers.LocalObject host);
-      end;
+      let source_host = Db.VM.get_resident_on ~__context ~self:vm in
+      let to_equal_or_greater_version = Helpers.host_versions_not_decreasing ~__context
+        ~host_from:(Helpers.LocalObject source_host)
+        ~host_to:(Helpers.LocalObject host) in
+
+      if (Helpers.rolling_upgrade_in_progress ~__context) && (not to_equal_or_greater_version) then
```

Comment:
The parentheses could be dropped in the `if` expression:

    Helpers.rolling_upgrade_in_progress ~__context && not to_equal_or_greater_version

Function application has higher precedence than `&&`.